tests/phpunit/elementor/core/page-assets/test-loader.php

```diff
@@ -0,0 +1,202 @@
+use Elementor\Testing\Elementor_Test_Base;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+class Test_Loader extends Elementor_Test_Base {
+	public function test_init_assets() {
+		$assets_loader = $this->elementor()->assets_loader;
+
+		$assets = $assets_loader->get_assets();
+
+		$this->assertArrayHasKey( 'styles', $assets, 'styles key was not initialized.' );
+		$this->assertArrayHasKey( 'scripts', $assets, 'scripts key was not initialized.' );
+	}
+
+	public function test_get_assets() {
+		$assets_loader = $this->elementor()->assets_loader;
+
+		$this->add_testing_assets( 'e-style-get-asset', 'e-script-get-asset' );
+
+		$assets = $assets_loader->get_assets();
+
+		$this->assertArrayHasKey( 'e-style-get-asset', $assets['styles'], 'Can not retrieve e-style-get-asset from assets.' );
+		$this->assertArrayHasKey( 'e-script-get-asset', $assets['scripts'], 'Can not retrieve e-script-get-asset from assets.' );
+	}
+
+	public function test_enable_asset() {
+		$assets_loader = $this->elementor()->assets_loader;
+
+		$this->add_testing_assets( 'e-style-enable-asset', 'e-script-enable-asset' );
+
+		$assets = $assets_loader->get_assets();
+
+		$this->assertArrayNotHasKey( 'enabled', $assets['styles']['e-style-enable-asset'], 'e-style-enable-asset should not be enabled by default.' );
+		$this->assertArrayNotHasKey( 'enabled', $assets['scripts']['e-script-enable-asset'], 'e-script-enable-asset should not be enabled by default.' );
+
+		$assets_loader->enable_assets( [
+			'styles' => ['e-style-enable-asset'],
+			'scripts' => ['e-script-enable-asset'],
+		] );
+
+		$assets = $assets_loader->get_assets();
+
+		$this->assertArrayHasKey( 'enabled', $assets['styles']['e-style-enable-asset'], 'e-style-enable-asset was not properly enabled.' );
+		$this->assertArrayHasKey( 'enabled', $assets['scripts']['e-script-enable-asset'], 'e-script-enable-asset was not properly enabled.' );
+	}
+
+	public function test_add_assets() {
+		$assets_loader = $this->elementor()->assets_loader;
+
+		$assets = [
+			'styles' => [
+				'e-style-add-asset' => [
+					'src' => 'e-style-add-asset-src',
+					'version' => 'e-style-add-asset-version',
+					'dependencies' => [],
+				],
+			],
+			'scripts' => [
+				'e-script-add-asset' => [
+					'src' => 'e-script-add-asset-src',
+					'version' => 'e-script-add-asset-version',
+					'dependencies' => [],
+				],
+			],
+		];
+
+		$assets_loader->add_assets( $assets );
+
+		$added_assets = $assets_loader->get_assets();
+
+		foreach ( $assets as $assets_type => $assets_type_data ) {
+			$this->assertArrayHasKey( $assets_type, $added_assets, $assets_type . ' was not properly added to assets.' );
+
+			foreach ( $assets_type_data as $asset_name => $asset_data ) {
+				$this->assertArrayHasKey( $asset_name, $added_assets[ $assets_type ], $asset_name . ' was not properly added to ' . $assets_type . ' of assets.' );
+
+				foreach( $asset_data as $key => $value ) {
```

Comment:
Compare the whole array instead of separated data values

Reply:
Done.